pkg/utils/cipher.go

```diff
@@ -55,12 +55,87 @@ func CryptoConfigWrapper(encBlock *evecommon.EncryptionBlock, cmnCryptoCfg *Comm
 
 	mEncBlock, mErr := proto.Marshal(encBlock)
 	if mErr != nil {
-		return nil, fmt.Errorf("error marshalling user data: %v", mErr)
+		return nil, fmt.Errorf("error marshalling user data: %w", mErr)
 	}
-	//Fill CipherBlock.
+	// Fill CipherBlock.
 	cipherBlock, cbErr := createCipherBlock(mEncBlock, cipherCtx.ContextId, cmnCryptoCfg, iv[:16])
 	if cbErr != nil {
-		return nil, fmt.Errorf("error creating cipher block: %v", cbErr)
+		return nil, fmt.Errorf("error creating cipher block: %w", cbErr)
 	}
 	return cipherBlock, nil
 }
+
+// internal decryption method
+func aesDecrypt(iv, symmetricKey, ciphertext []byte) ([]byte, error) {
+	plaintext := make([]byte, len(ciphertext))
+	aesBlockDecrypter, err := aes.NewCipher(symmetricKey)
+	if err != nil {
+		return nil, err
+	}
+	aesDecrypter := cipher.NewCFBDecrypter(aesBlockDecrypter, iv)
+	aesDecrypter.XORKeyStream(plaintext, ciphertext)
+	return plaintext, nil
+}
+
+// reverse the process of creating a cipher block
+func decryptCipherBlock(cipherBlock *evecommon.CipherBlock, cmnCryptoCfg *CommonCryptoConfig) ([]byte, error) {
+	if cipherBlock == nil {
+		return nil, fmt.Errorf("nil cipher block in decrypt cipher block method")
+	}
+
+	// decrypt ciphertext using symmetric key and initial value
+	decryptedText, decErr := aesDecrypt(cipherBlock.InitialValue, cmnCryptoCfg.SymmetricKey, cipherBlock.CipherData)
+	if decErr != nil {
+		return nil, decErr
+	}
+
+	// verify sha256 checksum of decrypted text
+	shaOfDecryptedText := sha256.Sum256(decryptedText)
+	if !CompareSlices(shaOfDecryptedText[:], cipherBlock.ClearTextSha256) {
```

Comment:
Curious, ill `bytes.Equal` work here?
Well, looks like the [logic](https://cs.opensource.google/go/go/+/release-branch.go1.20:src/bytes/bytes.go;drc=dcb90152a444be97fcc45bb12d176641c1b0d90e;l=20) inside of library function is suboptimal. So why to not use generics.